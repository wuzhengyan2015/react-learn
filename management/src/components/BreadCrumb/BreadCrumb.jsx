import React, { Component } from 'react'

/* eslint-disable */
class BreadCrumb extends Component {
  render() {
    return (
      <div>
        state
      </div>
    )
  }
}

export default BreadCrumb