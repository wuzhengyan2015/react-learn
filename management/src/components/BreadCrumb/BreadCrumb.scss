.ui-breadcrumb {
  height: 30px;
  line-height: 30px;
  margin-bottom: 16px;
  background: #f8f8f8;
}