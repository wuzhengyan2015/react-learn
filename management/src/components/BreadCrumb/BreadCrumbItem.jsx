import React, { Component } from 'react'
import './BreadCrumbItem.scss'

/* eslint-disable */
export default class BreadCrumbItem extends Component {
  render() {
    return (
      <div className="ui-breadcrumb-item">
        { this.props.children }
      </div>
    )
  }
}
