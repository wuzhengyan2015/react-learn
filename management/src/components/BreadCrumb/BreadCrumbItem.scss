.ui-breadcrumb-item {
  display: inline-block;
}