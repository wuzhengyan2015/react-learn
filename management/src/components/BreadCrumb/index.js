import BreadCrumb from './BreadCrumb'
import BreadCrumbItem from './BreadCrumbItem'

BreadCrumb.item = BreadCrumbItem

export default BreadCrumb
