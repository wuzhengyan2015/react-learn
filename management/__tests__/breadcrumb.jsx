import React from 'react'
import renderer from 'react-test-renderer'
import BreadCrumb from '../src/components/BreadCrumb/index'
import Enzyme, {shallow, mountm, render } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

const Item = BreadCrumb.item
Enzyme.configure({
  adapter: new Adapter(),
})

it('面包屑的渲染', () => {
  const tree = renderer
    .create(<BreadCrumb><Item>First</Item><Item>Second</Item></BreadCrumb>)
    .toJSON(); 
  expect(tree).toMatchSnapshot();
})

/* test('面包屑文字', () => {
  const breadcrumb = render(<BreadCrumb><Item>First</Item><Item>Second</Item></BreadCrumb>)
  expect(breadcrumb.find('div')).toEqual('First')
}) */
class CustomError extends Error {}
const binaryStringToNumber = binString => {
  if (!/^[01]+$/.test(binString)) {
    throw new CustomError('Not a binary number.');
  }

  return parseInt(binString, 2);
};

describe('binaryStringToNumber', () => {
  describe('given an invalid binary string', () => {
    test('composed of non-numbers throws CustomError', () => {
      expect(() => binaryStringToNumber('abc')).toThrow('Not a binary number.');
    });

    test('with extra whitespace throws CustomError', () => {
      expect(() => binaryStringToNumber('  100')).toThrow('Not a binary number.');
    });
  });

  describe('given a valid binary string', () => {
    test('returns the correct number', () => {
      expect(binaryStringToNumber('100')).toBe(4);
    });
  });
});

