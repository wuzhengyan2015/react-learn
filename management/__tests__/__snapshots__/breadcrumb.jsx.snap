// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`面包屑的渲染 1`] = `
<div
  className="ui-breadcrumb"
>
  <div
    className="ui-breadcrumb-item"
  >
    First
  </div>
  <div
    className="ui-breadcrumb-item"
  >
    Second
  </div>
</div>
`;
